functests/ipv6/ipv6.go

```diff
@@ -48,35 +51,62 @@ var _ = Describe("sriov", func() {
 	})
 
 	var _ = Context("IPv6 configured secondary interfaces on pods", func() {
-		It("Should be able to ping each other", func() {
-			podDefinition := dualStackTestPod(pingedPodName, []string{"/bin/bash", "-c", "--"}, []string{"while true; do sleep 300000; done;"}, nodeSelector)
-			_, err := client.Client.Pods(testNamespace).Create(podDefinition)
-			Eventually(func() k8sv1.PodPhase {
-				runningPod, err := client.Client.Pods(testNamespace).Get(pingedPodName, metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				return runningPod.Status.Phase
-			}, 3*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodRunning))
-			pod, err := client.Client.Pods(testNamespace).Get(pingedPodName, metav1.GetOptions{})
-			var nets []Network
-			err = json.Unmarshal([]byte(pod.ObjectMeta.Annotations["k8s.v1.cni.cncf.io/networks-status"]), &nets)
-			Expect(err).ToNot(HaveOccurred())
-			for _, net := range nets {
-				if net.Interface != sriovInterfaceName {
-					continue
-				}
-				for i, ip := range net.Ips {
-					pingPod(ip, i, nodeSelector)
-				}
-				return
+		It("Should be able to ping each other in a dual stack configuration", func() {
+
+			pod := createTestPod(dualStackPingedPodName, []string{"/bin/bash", "-c", "--"},
+				[]string{"while true; do sleep 300000; done;"}, nodeSelector, dualStackSriovNetworkAttachment)
+			ips := getSriovNicIps(pod)
+			Expect(len(ips)).Should(Equal(2))
+			for i, ip := range ips {
+				pingPod(dualStackPingingPodName, ip, i, nodeSelector, dualStackSriovNetworkAttachment)
+			}
+		})
+
+		It("Should be able to ping each other in a IPv6 only configuration", func() {
+
+			pod := createTestPod(ipv6PingedPodName, []string{"/bin/bash", "-c", "--"},
+				[]string{"while true; do sleep 300000; done;"}, nodeSelector, ipv6onlySriovNetworkAttachment)
+			ips := getSriovNicIps(pod)
+			Expect(len(ips)).Should(Equal(1))
+			for i, ip := range ips {
+				pingPod(ipv6PingingPodName, ip, i, nodeSelector, ipv6onlySriovNetworkAttachment)
 			}
-			Fail("No sriov network inerface found.")
 		})
 	})
 })
 
-func pingPod(ip string, podNumber int, nodeSelector string) {
-	podName := fmt.Sprintf("%s-%d", pingingPodName, podNumber)
-	podDefinition := dualStackTestPod(podName, []string{"sh", "-c", "ping -c 3 " + ip}, []string{}, nodeSelector)
+func getSriovNicIps(pod *k8sv1.Pod) []string {
+	var nets []Network
+	err := json.Unmarshal([]byte(pod.ObjectMeta.Annotations["k8s.v1.cni.cncf.io/networks-status"]), &nets)
+	Expect(err).ToNot(HaveOccurred())
+	for _, net := range nets {
+		if net.Interface != sriovInterfaceName {
+			continue
+		}
+		return net.Ips
+	}
+	Fail("No sriov network inerface found.")
```

Comment:
Can we check the len & eventually fail outside?
This makes this function more flexible and not depending on the gingko package.

Reply:
Done